packages/local_auth/local_auth_darwin/darwin/Classes/FLALocalAuthPlugin.m

```diff
@@ -130,12 +132,31 @@ - (nullable NSNumber *)deviceCanSupportBiometricsWithError:
   if ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
                            error:&authError]) {
     if (authError == nil) {
+#if TARGET_OS_OSX
+      // @available(macOS 10.15, *) is not a BOOL expression.
+      // It does not return an indication of whether the code is being run on that version of iOS or
+      // not. So there needs to be a separate check for the macOS version and on the biometry type
+      // https://stackoverflow.com/questions/52251973/ios-available-does-not-guard-availability-here-use-if-available-instead
+      if (@available(macOS 10.15, *)) {
+        if (context.biometryType == LABiometryTypeFaceID) {
+          [biometrics addObject:[FLADAuthBiometricWrapper makeWithValue:FLADAuthBiometricFace]];
+          return biometrics;
+        }
+      }
+      if (context.biometryType == LABiometryTypeTouchID) {
+        [biometrics
+            addObject:[FLADAuthBiometricWrapper makeWithValue:FLADAuthBiometricFingerprint]];
+      }
+#endif
+
+#if TARGET_OS_IOS
```

Comment:
Why is all of this code duplicated and ifdef'd rather than the `@available` check including iOS?

Reply:
When I try to combine both availability statements I get this error:
```
 @available does not guard availability here; use if (@available) instead [-Wunsupported-availability-guard]
       if (@available(macOS 10.15, *) || @available(iOS 11.0, *)) {
```

That was why I put the comment saying @available is not a BOOl expression and is why I had to separate the if statements. What would you suggest we do then?

Reply:
Nvm fixed: https://forums.developer.apple.com/forums/thread/5237